backend/supabase/contracts/empty-search-contracts-function.sql

```diff
@@ -0,0 +1,84 @@
+CREATE OR REPLACE FUNCTION empty_search_contracts(
+    contract_filter TEXT,
+    contract_sort TEXT,
+    offset_n INTEGER,
+    limit_n INTEGER,
+    group_id TEXT DEFAULT NULL,
+    creator_id TEXT DEFAULT NULL
+  ) RETURNS TABLE (data jsonb) AS $$
+DECLARE base_query TEXT;
+where_clause TEXT;
+sql_query TEXT;
+BEGIN -- Common WHERE clause
+-- If fuzzy search is enabled and is group search
+IF group_id is not null then base_query := FORMAT(
```

Comment:
I'm not sure about generating the sql from the sql function. It's maybe ok if we're just doing a small amount of it. But it seems like we'd much prefer creating all the sort, where clause, and fuzzy vs full text search from a typescript api endpoint.

Any reason why not to do this?